### DISCUSSION TITLE
Implement Shield Mode

### DISCUSSION THREAD

**Discussion by nmussy** [Chatterino/chatterino2#4206] [Ideas]
The [newly added Shield Mode](https://safety.twitch.tv/s/article/Protect-your-channel-with-Shield-Mode) would be a great addition, alongside the other mod controls on the top right of the chat window. It can be toggled via two chat commands, `/shield` and `/shieldoff` :

![image](https://user-images.githubusercontent.com/2505696/205051102-3396f0da-14d6-4501-8c40-6b9462d6ffaa.png)

**Comment by LosFarmosCTL** [comment]
At least afaik Twitch does not provide an API for that, so currently not possible for chatterino to implement.

**Comment by nmussy** [comment]
Apologies, I assumed they would have rolled it out on the API at the same time, but it's apparently not yet part of the [`/chat/settings` endpoint](https://dev.twitch.tv/docs/api/reference#update-chat-settings)

**Comment by Nerixyz** [comment]
~~[Relevant uservoice](https://twitch.uservoice.com/forums/933812-safety/suggestions/46033573-shield-mode-api-and-bot-protection).~~ [API Docs for Shield Mode](https://dev.twitch.tv/docs/api/reference#update-shield-mode-status).

**Comment by LosFarmosCTL** [comment]
Twitch has officially announced new [Helix endpoints as well as EventSub types](https://discuss.dev.twitch.tv/t/new-shield-mode-endpoints-and-eventsub-subscription-type-are-now-in-open-beta/42172/1) for managing and displaying Shield Mode status.

**Reply by Nerixyz** [reply]
I don't see a PubSub topic there. It says only EventSub subscriptions (and Helix endpoints) were added.

**Reply by LosFarmosCTL** [reply]
I should not write anything at 6am 🫠
